src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -954,25 +863,10 @@ public SecurityComponents createSecurityComponents() {
         return new SecurityComponents(new AuthenticationManager() {
             public Authentication authenticate(Authentication authentication) throws AuthenticationException {
                 if (authentication instanceof AnonymousAuthenticationToken) return authentication;
-
-                if (authentication instanceof UsernamePasswordAuthenticationToken && escapeHatchEnabled) {
-                    randomWait(); // to slowdown brute forcing
-                    if (checkEscapeHatch(
-                            authentication.getPrincipal().toString(),
-                            authentication.getCredentials().toString())) {
-                        List<GrantedAuthority> grantedAuthorities = new ArrayList<>();
-                        grantedAuthorities.add(SecurityRealm.AUTHENTICATED_AUTHORITY2);
-                        if (isNotBlank(escapeHatchGroup)) {
-                            grantedAuthorities.add(new SimpleGrantedAuthority(escapeHatchGroup));
-                        }
-                        UsernamePasswordAuthenticationToken token =
-                                new UsernamePasswordAuthenticationToken(escapeHatchUsername, "", grantedAuthorities);
-                        SecurityContextHolder.getContext().setAuthentication(token);
-                        OicUserDetails userDetails = new OicUserDetails(escapeHatchUsername, grantedAuthorities);
-                        SecurityListener.fireAuthenticated2(userDetails);
-                        return token;
-                    } else {
-                        throw new BadCredentialsException("Wrong username and password: " + authentication);
+                for (var property : properties) {
+                    var authenticate = property.authenticate(authentication);
```

Comment:
if one property throws an exception (which is what `EscapeHatch` does) how would another property provide an authentication?
in a bizzare scenario where the escap hatch used the same username but a different password than something in some other Property?

does this need to catch the exception and try all others, (and if they throw exceptions keep adding a suppresed exception to the original), then if non pass throw the exception?

Reply:
> bizzare scenario

Seems very unlikely. Why would you need to define multiple ways to log in using user/password ? I could imagine a plugin contributing a more complex implementation of escape hatch, but in this case why would you configure both?